[CI] EsqlAsyncSecurityIT testDatasetCrudForbiddenWithoutIndexManage failing

**Build Scans:**
- [elasticsearch-periodic #12222 / release-tests](https://gradle-enterprise.elastic.co/s/fdbwsabu5reoi)
- [elasticsearch-periodic #12222 / release-tests](https://gradle-enterprise.elastic.co/s/ipjyhgj5xsmtk)
- [elasticsearch-periodic-java-ea #256 / release-tests](https://gradle-enterprise.elastic.co/s/og5v32pdcvauc)
- [elasticsearch-periodic #12217 / release-tests](https://gradle-enterprise.elastic.co/s/xeghbqc7a4qmi)
- [elasticsearch-periodic #12217 / release-tests](https://gradle-enterprise.elastic.co/s/qyjga5vedxz7y)
- [elasticsearch-periodic-java-ea #255 / release-tests](https://gradle-enterprise.elastic.co/s/yqp4sbvuxglra)
- [elasticsearch-pull-request #142304 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/ypiqsjj26fdzu)
- [elasticsearch-pull-request #142304 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/4xtomdzkt6gha)
- [elasticsearch-periodic-java-ea #254 / release-tests](https://gradle-enterprise.elastic.co/s/j5r6xe3t2m6us)
- [elasticsearch-pull-request #142124 / checkpart4_release-tests](https://gradle-enterprise.elastic.co/s/auhoplxa54pvo)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:security:javaRestTest" --tests "org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT.testDatasetCrudForbiddenWithoutIndexManage" -Dtests.seed=B5F7B9F5559D4A7B -Dbuild.snapshot=false -Dtests.jvm.argline="-Dbuild.snapshot=false" -Dlicense.key=x-pack/license-tools/src/test/resources/public.key -Dtests.locale=vec-IT -Dtests.timezone=Antarctica/Troll -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testDatasetCrudForbiddenWithoutIndexManage'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <403>
     but: was <400>
```

**Issue Reasons:**
- [main] 9 consecutive failures in step release-tests
- [main] 16 consecutive failures in step checkpart4_release-tests
- [main] 25 failures in test testDatasetCrudForbiddenWithoutIndexManage (2.7% fail rate in 936 executions)
- [main] 9 failures in step release-tests (100.0% fail rate in 9 executions)
- [main] 16 failures in step checkpart4_release-tests (100.0% fail rate in 16 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (40.0% fail rate in 5 executions)
- [main] 8 failures in pipeline elasticsearch-pull-request (1.5% fail rate in 533 executions)
- [main] 5 failures in pipeline elasticsearch-periodic-java-ea (100.0% fail rate in 5 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,24 +354,24 @@ tests:
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT
   method: testTermVectorsApiRealtimeGet
   issue: https://github.com/elastic/elasticsearch/issues/148175
-- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-  method: testDatasetCrudForbiddenWithoutIndexManage
-  issue: https://github.com/elastic/elasticsearch/issues/148182
-- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-  method: testDataSourceAdminListEmpty
-  issue: https://github.com/elastic/elasticsearch/issues/148183
-- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-  method: testDataSourceCrudForbiddenWithoutClusterManage
-  issue: https://github.com/elastic/elasticsearch/issues/148184
-- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-  method: testDatasetCrudForbiddenWithoutIndexManage
-  issue: https://github.com/elastic/elasticsearch/issues/148185
-- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-  method: testDataSourceAdminListEmpty
-  issue: https://github.com/elastic/elasticsearch/issues/148186
-- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-  method: testDataSourceCrudForbiddenWithoutClusterManage
-  issue: https://github.com/elastic/elasticsearch/issues/148187
+#- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
+#  method: testDatasetCrudForbiddenWithoutIndexManage
+#  issue: https://github.com/elastic/elasticsearch/issues/148182
+#- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
+#  method: testDataSourceAdminListEmpty
+#  issue: https://github.com/elastic/elasticsearch/issues/148183
+#- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
+#  method: testDataSourceCrudForbiddenWithoutClusterManage
+#  issue: https://github.com/elastic/elasticsearch/issues/148184
+#- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
+#  method: testDatasetCrudForbiddenWithoutIndexManage
+#  issue: https://github.com/elastic/elasticsearch/issues/148185
+#- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
+#  method: testDataSourceAdminListEmpty
+#  issue: https://github.com/elastic/elasticsearch/issues/148186
+#- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
+#  method: testDataSourceCrudForbiddenWithoutClusterManage
+#  issue: https://github.com/elastic/elasticsearch/issues/148187
 - class: org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests
   method: testCreateAndRestoreSearchableSnapshot
   issue: https://github.com/elastic/elasticsearch/issues/148217

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -354,24 +354,6 @@ tests:
 - class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT
   method: testTermVectorsApiRealtimeGet
   issue: https://github.com/elastic/elasticsearch/issues/148175
-#- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-#  method: testDatasetCrudForbiddenWithoutIndexManage
-#  issue: https://github.com/elastic/elasticsearch/issues/148182
-#- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-#  method: testDataSourceAdminListEmpty
-#  issue: https://github.com/elastic/elasticsearch/issues/148183
-#- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-#  method: testDataSourceCrudForbiddenWithoutClusterManage
-#  issue: https://github.com/elastic/elasticsearch/issues/148184
-#- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-#  method: testDatasetCrudForbiddenWithoutIndexManage
-#  issue: https://github.com/elastic/elasticsearch/issues/148185
-#- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-#  method: testDataSourceAdminListEmpty
-#  issue: https://github.com/elastic/elasticsearch/issues/148186
-#- class: org.elasticsearch.xpack.esql.EsqlSecurityIT
-#  method: testDataSourceCrudForbiddenWithoutClusterManage
-#  issue: https://github.com/elastic/elasticsearch/issues/148187
 - class: org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsIntegTests
   method: testCreateAndRestoreSearchableSnapshot
   issue: https://github.com/elastic/elasticsearch/issues/148217

--- a/x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java
+++ b/x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java
@@ -2012,7 +2012,8 @@ public class EsqlSecurityIT extends ESRestTestCase {
     }
 
     // TODO: use named privileges when available — https://github.com/elastic/elasticsearch/issues/147017
-    public void testDataSourceCrudForbiddenWithoutClusterManage() {
+    public void testDataSourceCrudForbiddenWithoutClusterManage() throws IOException {
+        assumeTrue("data_sources REST API not supported by cluster", dataSourcesApiSupported());
         // user2 has cluster: []. All three data source actions are cluster-level → 403.
         for (String path : List.of("/_query/data_source/ds_x", "/_query/data_source")) {
             Request get = new Request("GET", path);
@@ -2033,7 +2034,8 @@ public class EsqlSecurityIT extends ESRestTestCase {
         assertThat(ex.getResponse().getStatusLine().getStatusCode(), equalTo(403));
     }
 
-    public void testDatasetCrudForbiddenWithoutIndexManage() {
+    public void testDatasetCrudForbiddenWithoutIndexManage() throws IOException {
+        assumeTrue("data_sources REST API not supported by cluster", dataSourcesApiSupported());
         // user3 has only `read` on `index`. Dataset actions need index `manage`.
         Request put = new Request("PUT", "/_query/dataset/index");
         put.setJsonEntity("{\"data_source\":\"parent\",\"resource\":\"s3://b/\"}");

